geode-core/src/main/java/org/apache/geode/cache/query/Index.java

```diff
@@ -115,4 +115,18 @@ default boolean isValid() {
    * @return the projectionAttributes, or "*" if there were none specified at index creation.
    */
   String getCanonicalizedProjectionAttributes();
+
+  /**
+   * Set whether maintenance of index is ongoing.
+   */
+  default void setMaintenanceOngoing(boolean b) {
```

Comment:
I don't think adding a public mutator API to this interface is correct.  What happens when a user application invokes this method?